### ISSUE TITLE
target 'rules_python_wheel_entry_point_hjson' not declared in package

### ISSUE THREAD

**Issue by kriswuollett** [bazelbuild/rules_python#602]
# 🐞 bug report

The `entry_point` for package `hjson` does not work with `v0.6.0`.

Is `entry_point` not fully available for use yet? I don't see any mention of it on the top level `README.md`.

### Affected Rule

`entry_point` method

### Is this a regression?

Not sure. First time I've tried using this feature of `rules_python`. Perhaps this package is a different case of an invalid entry points config?

### Description

The `rules_python_wheel_entry_point_hjson` not generated for requirement [`hjson==3.0.2`](https://pypi.org/project/hjson/3.0.2/):

## 🔬 Minimal Reproduction

* [sample workspace in `rules_python` branch](https://github.com/kriswuollett/rules_python/commit/5d56282371f1e80c4d170a940d3f4ba93f71397b), try running any of the targets like `bazel run //:hjson`.

## 🔥 Exception or Error
```
bazel run //:hjson                  
INFO: Invocation ID: 1389b1ea-dff0-4c15-8cc2-1d0a248ceb3c
ERROR: /Users/kris/workspace/github.com/kriswuollett/rules_python/examples/entrypoint/BUILD:3:6: no such target '@pip_deps_hjson//:rules_python_wheel_entry_point_hjson': target 'rules_python_wheel_entry_point_hjson' not declared in package '' defined by /private/var/tmp/_bazel_kris/0e05fb82d063bbe0995a0b2b910a0be4/external/pip_deps_hjson/BUILD.bazel and referenced by '//:hjson'
ERROR: Analysis of target '//:hjson' failed; build aborted: Analysis failed
INFO: Elapsed time: 0.114s
INFO: 0 processes.
FAILED: Build did NOT complete successfully (0 packages loaded, 0 targets configured)
FAILED: Build did NOT complete successfully (0 packages loaded, 0 targets configured)
```

## 🌍  Your Environment

**Operating System:**

<pre>
  <code>
macOS 12.1
  </code>
</pre>

**Output of `bazel version`:**

<pre>
  <code>
Bazelisk version: development
Build label: 4.2.2
Build target: bazel-out/darwin-opt/bin/src/main/java/com/google/devtools/build/lib/bazel/BazelServer_deploy.jar
Build time: Thu Dec 2 18:28:52 2021 (1638469732)
Build timestamp: 1638469732
Build timestamp as int: 1638469732
  </code>
</pre>

**Rules_python version:**

<pre>
  <code>
http_archive(
    name = "rules_python",
    sha256 = "a30abdfc7126d497a7698c29c46ea9901c6392d6ed315171a6df5ce433aa4502",
    strip_prefix = "rules_python-0.6.0",
    url = "https://github.com/bazelbuild/rules_python/archive/0.6.0.tar.gz",
)

  </code>
</pre>

**Comment by UebelAndre** [comment]
This issue is caused by the `hjson` executable coming from setup.py, and not being a console script (entry point) defined in the wheel.

https://github.com/hjson/hjson-py/blob/master/setup.py#L75

This could potentially be fixed by https://github.com/bazelbuild/rules_python/pull/575

**Comment by UebelAndre** [comment]
Cc @groodt 
This would be fixed by having support for setup.py scripts. Perhaps this can be collapsed or covered to a ticket tracking that specific feature?

**Comment by groodt** [comment]
Agreed. I think I'll create an "umbrella" tracking issue for "legacy scripts support".

fwiw, it seems that perhaps hsjon will switch to entrypoints (or at least are being encouraged by conda to do so) https://github.com/hjson/hjson-py/pull/29

**Comment by groodt** [comment]
This particular issue with hjson is fixed in `hjson==3.1.0`.

Tested against the minimal reproduction and it works.

```
❯ ../../bazel-5.2.0-darwin-x86_64 run //:hjson -- --help
INFO: Analyzed target //:hjson (0 packages loaded, 0 targets configured).
INFO: Found 1 target...
Target @pip_deps_hjson//:rules_python_wheel_entry_point_hjson up-to-date:
  bazel-bin/external/pip_deps_hjson/rules_python_wheel_entry_point_hjson
INFO: Elapsed time: 0.201s, Critical Path: 0.00s
INFO: 1 process: 1 internal.
INFO: Build completed successfully, 1 total action
INFO: Build completed successfully, 1 total action
Hjson, a user interface for JSON

Usage:
  hjson [options]
  hjson [options] <input>
  hjson (-h | --help)
  hjson (-V | --version)

Options:
  -h --help     Show this screen.
  -j            Output as formatted JSON.
  -c            Output as JSON.
  -V --version  Show version.

Hjson, a user interface for JSON

Usage:
  hjson [options]
  hjson [options] <input>
  hjson (-h | --help)
  hjson (-V | --version)

Options:
  -h --help     Show this screen.
  -j            Output as formatted JSON.
  -c            Output as JSON.
  -V --version  Show version.
```

Closing.
